src/background_workers.c

```diff
@@ -146,30 +208,81 @@ void *RedisAI_Run_ThreadMain(void *arg) {
         evicted_items = array_append(evicted_items, item);
         batch_rinfo = array_append(batch_rinfo, rinfo);
 
-        // If the item is a scriptrun, we stop looking for matching items to
-        // batch because there's no batching for script
-        if (rinfo->sctx) {
+        // Get the currentOp from the DAG at the top of the queue
+        // The currentOp is the first op without a result that needs to run
+        // on the device for the queue
+        // Also collect info on:
+        // - whether the currentOp is ready (all its inputs
+        //   are present in the context)
+        // - whether the currentOp is batchable (that is, it is a model and
+        //   it was set with batchsize > 0)
+        RAI_DagOp* currentOp = RedisAI_DagCurrentOp(rinfo);
+
+        // Have all ops that need to run on the device been executed
+        int deviceComplete = RedisAI_DagDeviceComplete(rinfo);
```

Comment:
I think you can infer this one at the end

Reply:
At this point, after the last changes, it is very lightweight to just get the information. We need this info to short-circuit the other operations in the worker loop, so I'd rather keep this here.